src/data/mv_polynomial/equiv.lean

```diff
@@ -319,6 +321,258 @@ begin
+              polynomial.eval_mul],
+    congr,
+    { exact h },
+    { clear h f,
+      simp only [fin_succ_equiv_apply, eval₂_hom_X'],
+      by_cases c : j = 0,
+      { rw c,
+        simp [fin.cons_zero] },
+      { have c' : j ≠ 0 := by simpa only [ne.def],
+        rw [←fin.succ_pred j c', fin.cons_succ],
+        simp } } },
+end
+
+lemma coeff_eval_eq_eval_coeff {n : ℕ} (s' : fin n → R) (f : polynomial (mv_polynomial (fin n) R))
+  (i : ℕ) : polynomial.coeff (polynomial.map (eval s') f) i =  eval s' (polynomial.coeff f i) :=
+by simp only [polynomial.coeff_map]
+
+/- TODO: what is a good name for this one? -/
+lemma support_eval {n : ℕ} (s' : fin n → R)
+  (f : polynomial (mv_polynomial (fin n) R)) : (polynomial.map (eval s') f).support ⊆ f.support :=
+begin
+  intros i hi,
+  simp only [polynomial.mem_support_iff, polynomial.coeff_map, ne.def] at hi,
+  by_contradiction c,
+  simp only [polynomial.mem_support_iff, not_not, ne.def] at c,
+  rw c at hi,
+  simpa using hi,
+end
+
+lemma degree_eval_le_degree {n : ℕ} (s' : fin n → R) (f : polynomial (mv_polynomial (fin n) R)) :
+  polynomial.degree (polynomial.map (eval s') f) ≤ polynomial.degree f :=
+polynomial.degree_mono (support_eval s' f)
+
+lemma nat_degree_eval_le_nat_degree {n : ℕ} (s : fin n → R)
+  (f : polynomial (mv_polynomial (fin n) R)) :
+  polynomial.nat_degree (polynomial.map (eval s) f) ≤ polynomial.nat_degree f :=
+polynomial.nat_degree_le_nat_degree (degree_eval_le_degree s f)
+
+lemma support_coeff_fin_succ_equiv {n : ℕ} {f : mv_polynomial (fin (n + 1)) R} {i : ℕ}
+  {m : fin n →₀ ℕ } : m ∈ (polynomial.coeff ((fin_succ_equiv R n) f) i).support
+   ↔ (finsupp.cons i m) ∈ f.support :=
+begin
+  apply iff.intro,
+  { intro h,
+    simpa [←fin_succ_equiv_coeff_coeff] using h },
+  { intro h,
+    simpa [mem_support_iff, ←fin_succ_equiv_coeff_coeff m f i] using h },
+end
+
+lemma fin_succ_equiv_support {n : ℕ} (f : mv_polynomial (fin (n + 1)) R) :
+  (fin_succ_equiv R n f).support = finset.image (λ m : fin (n + 1)→₀ ℕ, m 0) f.support :=
+begin
+  apply finset.subset.antisymm,
+  { intros i hi,
+    rw finset.mem_image,
+    rw [polynomial.mem_support_iff, nonzero_iff_exists] at hi,
+    cases hi with m hm,
+    use cons i m,
+    apply and.intro,
+    { rw ← support_coeff_fin_succ_equiv,
+      simpa using hm },
+    { rw cons_zero } },
+  { intros i hi,
+    rw polynomial.mem_support_iff,
+    rw finset.mem_image at hi,
+    cases hi with m hm,
+    cases hm with h hm,
+    rw nonzero_iff_exists,
+    use tail m,
+    rwa [← coeff, ← mem_support_iff, support_coeff_fin_succ_equiv, ← hm, cons_tail] },
+end
+
+lemma fin_succ_equiv_support' {n : ℕ} {f : mv_polynomial (fin (n + 1)) R} {i : ℕ} :
+  finset.image (finsupp.cons i) (polynomial.coeff ((fin_succ_equiv R n) f) i).support
+   = f.support.filter(λ m, m 0 = i) :=
+begin
+  ext m,
+  apply iff.intro,
+  { intro hm,
+    rw finset.mem_image at hm,
```

Comment:
Again, try to get these rewrites out of the way before splitting the goal

Reply:
Done! I had to use `conv` to move some of this rewrites. Is there a more reasonable way?